### PR TITLE
refactor and chore: addressing internal audit findings

### PR DESCRIPTION
**Pull request by kupermind**
- Adding recommended view functions in `wveOLAS` for the `ERC20` token;
- Adding tests to 100% coverage;
- Updating the audit report README to show addressed items.

### CODE REVIEW

**Review by 77ph — APPROVED** [review]

File: test/Deployment.js [hunk 1]
```diff
@@ -364,7 +364,7 @@ describe("Deployment", function () {
             // !!!!!!!!!!!!!!!!!!!!!!!! SUPPLEMENTAL DEPLOYMENT STEPS !!!!!!!!!!!!!!!!!!!!!!!!
             // 16. EOA to deploy wveOLAS contract pointed to veOLAS;
```
Comment (mariapiamo) [inline test/Deployment.js:365]:
```suggestion
            // 16. EOA to deploy wveOLAS contract pointed to veOLAS and OLAS;
```

File: contracts/wveOLAS.sol [hunk 2]
```diff
@@ -111,21 +111,40 @@ error WrongTimestamp(uint256 minTimeStamp, uint256 providedTimeStamp);
...
 /// @title wveOLAS - Wrapper smart contract for view functions of veOLAS contract
 /// @author AL
 /// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
 contract wveOLAS {
     // veOLAS address
     address public immutable ve;
+    // OLAS address
+    address public immutable token;
+    // veOLAS token name
+    string public constant name = "Voting Escrow OLAS";
```
Comment (mariapiamo) [inline contracts/wveOLAS.sol:131]: so we use the same name, we do not use the word wrapper here

File: contracts/wveOLAS.sol [hunk 3]
```diff
@@ -243,21 +261,54 @@ contract wveOLAS {
         vPower = IVEOLAS(ve).getPastTotalSupply(blockNumber);
     }
 
+    /// @dev Gets the total number of supply points.
+    /// @return numPoints Number of supply points.
+    function totalNumPoints() external view returns (uint256 numPoints) {
+        numPoints = IVEOLAS(ve).totalNumPoints();
+    }
+
```
Comment (mariapiamo) [inline contracts/wveOLAS.sol:269]: Why now we are doing this for `totalNumPoints()` and not for `mapSupplyPoints`?

File: contracts/wveOLAS.sol (hunk 2 above)
Comment (kupermind) [reply contracts/wveOLAS.sol:131]: no, as we need to return the veOLAS result at the end

File: contracts/wveOLAS.sol (hunk 3 above)
Comment (kupermind) [reply contracts/wveOLAS.sol:269]: They are just very custom, but ok let's add them as well

File: contracts/wveOLAS.sol [hunk 4]
```diff
@@ -100,32 +100,50 @@ interface IVEOLAS {
 }
 
 /// @dev Zero veOLAS address.
```
Comment (mariapiamo) [inline contracts/wveOLAS.sol:102]:
```suggestion
/// @dev Zero address.
```

**Review by mariapiamo — APPROVED** [review]

**Review by mariapiamo — APPROVED** [review]

**Pull request merged**